OpenRA.Game/Orders/UnitOrderGenerator.cs

```diff
@@ -112,27 +119,30 @@ static UnitOrderResult OrderForUnit(Actor self, Target target, MouseInput mi)
 			if (self.Disposed || !target.IsValidFor(self))
 				return null;
 
-			if (mi.Button == Game.Settings.Game.MouseButtonPreference.Action)
-			{
-				foreach (var o in self.TraitsImplementing<IIssueOrder>()
-					.SelectMany(trait => trait.Orders
-						.Select(x => new { Trait = trait, Order = x }))
-					.OrderByDescending(x => x.Order.OrderPriority))
-				{
-					var actorsAt = self.World.ActorMap.GetActorsAt(self.World.Map.CellContaining(target.CenterPosition)).ToList();
+			var modifiers = TargetModifiers.None;
+			if (mi.Modifiers.HasModifier(Modifiers.Ctrl))
+				modifiers |= TargetModifiers.ForceAttack;
+			if (mi.Modifiers.HasModifier(Modifiers.Shift))
+				modifiers |= TargetModifiers.ForceQueue;
+			if (mi.Modifiers.HasModifier(Modifiers.Alt))
+				modifiers |= TargetModifiers.ForceMove;
 
-					var modifiers = TargetModifiers.None;
-					if (mi.Modifiers.HasModifier(Modifiers.Ctrl))
-						modifiers |= TargetModifiers.ForceAttack;
-					if (mi.Modifiers.HasModifier(Modifiers.Shift))
-						modifiers |= TargetModifiers.ForceQueue;
-					if (mi.Modifiers.HasModifier(Modifiers.Alt))
-						modifiers |= TargetModifiers.ForceMove;
+			var orders = self.TraitsImplementing<IIssueOrder>()
+				.SelectMany(trait => trait.Orders.Select(x => new { Trait = trait, Order = x }))
+				.OrderByDescending(x => x.Order.OrderPriority);
```

Comment:
I'm not convinced: in the common case we only enumerate this once, and in the worst case we enumerate it twice.